### PR TITLE
Nx Feature - Certificates Management (List)

### PR DESCRIPTION
**Pull request by LeonamMoura**

### CODE REVIEW

File: src/js/views/Certificates/layout/AssociateDevicesModal.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  Button,
+  Radio,
+} from '@material-ui/core';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import { DialogHeader } from '../../../common/components/Dialogs';
+import { useDetailsModalStyles } from './style';
+
+const fakeRows = [
```
Comment (cfrancisco) [inline src/js/views/Certificates/layout/AssociateDevicesModal.jsx:23]: Fake code rsrs..

Comment (cfrancisco) [reply src/js/views/Certificates/layout/AssociateDevicesModal.jsx:23]: It's a good practice always to review our code after sending it as PR.

File: src/js/views/Certificates/translations/en.certificates.i18n.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "dataTableBody": {
+    "notAssociated": "Not associated"
+  },
+  "associateDeviceModal": {
+    "table": {
+      "id": "Id",
+      "devices": "Devices",
+      "lastUpdate": "Last update"
+    },
+    "associate": "Associar"
```
Comment (cfrancisco) [inline src/js/views/Certificates/translations/en.certificates.i18n.json:28]: Missing i18n

**Review by cfrancisco — CHANGES_REQUESTED** [review]

File: src/js/views/Certificates/View.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      <CertificateOptionsMenu
+        isShowingMenu={!!certificatesOptionsMenu}
+        anchorElement={certificatesOptionsMenu?.anchorElement}
+        certificate={certificatesOptionsMenu?.certificate}
+        handleHideOptionsMenu={handleHideOptionsMenu}
+        handleShowDevicesToAssociate={handleShowDevicesToAssociate}
+      />
+      <Box className={classes.container}>
+        <Box className={classes.content}>
+          <SearchBar />
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/View.jsx:50]: The SearchBar component receives some props to change the view mode and search certificates. You should pass these props.

File: src/js/views/Certificates/layout/Cards.jsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+import { Box, Grid, IconButton, Tooltip, Typography } from '@material-ui/core';
+import { Check, Close, CertificatesOther, Star, StarBorderOutlined } from '@material-ui/icons';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import DataCard from '../../../common/components/Cards/DataCard';
+import { useCardsStyles } from './style';
+
+const Cards = ({
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/layout/Cards.jsx:12]: You should remove everything related to devices, such as favorite buttons, attrs, and use this component in the `certificates/View.jsx` component.

File: src/js/views/Certificates/layout/EmptyDeviceList.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import { Box, Button, Grid, Typography } from '@material-ui/core';
+import { Add, CertificatesOther } from '@material-ui/icons';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router';
+
+const EmptyCertificateList = () => {
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/layout/EmptyDeviceList.jsx:8]: Use this component in the `certificates/View.jsx` component

File: src/js/views/Certificates/layout/MassActions.jsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+
+import { Grid, Box, Typography, Button, IconButton } from '@material-ui/core';
+import { VerifiedUser, Close, Delete, Star } from '@material-ui/icons';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import { useMassActionsStyles } from './style';
+
+const MassActions = ({
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/layout/MassActions.jsx:10]: The only button that should be on this component is delete all. Users cannot add certificates to favorites and the create certificates button is meaningless here.

File: src/js/views/Certificates/View.jsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+          <DataTable
+            rowsPerPage={10}
+            selectedCertificates={[]}
+            page={0}
+            certificates={[]}
+            handleSetCertificateOptionsMenu={setCertificatesOptionsMenu}
+            handleShowDevicesToAssociate={handleShowDevicesToAssociate}
+          />
+        </Box>
+        <Pagination />
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/View.jsx:60]: Missing props for the `<Pagination />` component

File: src/js/views/Certificates/layout/SearchBar.jsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        >
+          <ViewModule />
+        </IconButton>
+
+        <TextField
+          inputRef={searchInputRef}
+          className={classes.searchTextField}
+          size='small'
+          variant='outlined'
+          placeholder={t('searchInputPh')}
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/layout/SearchBar.jsx:73]: Create this placeholder in the translation files

**Review by LuanEdCosta — CHANGES_REQUESTED** [review]
Very good job! I found some small mistakes, but you're doing great, never stop learning :rocket:

File: src/js/views/Certificates/View.jsx [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+  const handleDeleteMultipleCertificates = () => {
+    setIsShowingMultipleDeleteAlert(true);
+  };
+
+  const handleDisassociateDevice = () => {
+    setIsShowingDisassociateDeviceAlert(true);
+  };
+
+  return (
+    <ViewContainer headerTitle='Certificados'>
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/View.jsx:129]: Missing translations

File: src/js/views/Certificates/View.jsx [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+          )}
+        </Box>
+
+        <Pagination
+          page={page}
+          rowsPerPage={rowsPerPage}
+          totalOfDevices={totalPages}
+          numberOfSelectedCertificates={selectedCertificates.length}
+          totalOfCertificates={certificates.length}
+          numberOfSelectedDevices={selectedCertificates.length}
```
Comment (LuanEdCosta) [inline src/js/views/Certificates/View.jsx:233]: I think you can remove this prop :thinking:

**Review by LuanEdCosta — CHANGES_REQUESTED** [review]
Everything is very good! :rocket: 
But I found two small mistakes :sweat_smile:

**Review by LuanEdCosta — APPROVED** [review]
:rocket: LGTM

**Review by cfrancisco — APPROVED** [review]

**Pull request merged**